File: app/src/test/java/com/originb/inkwisenote/modules/BackgroundOpsTest.java
package com.originb.inkwisenote.modules;

import com.originb.inkwisenote.modules.messaging.BackgroundOps;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class BackgroundOpsTest {

    @Test
    public void registerListener() {
        final int number = 0;
        BackgroundOps.execute(() -> {
            int newNumber = number + 1;
            assertTrue(newNumber == 2);
        });

        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

}

File: app/src/test/java/com/originb/inkwisenote2/modules/BackgroundOpsTest.kt
package com.originb.inkwisenote2.modules

import com.originb.inkwisenote2.modules.backgroundjobs.BackgroundOps.Companion.execute
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class BackgroundOpsTest {
    @Test
    fun registerListener() {
        val number = 0
        execute(Runnable {
            val newNumber = number + 1
            Assert.assertTrue(newNumber == 2)
        })

        try {
            Thread.sleep(5000)
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
    }
}
